clang/test/SemaCXX/linkage1.cpp

```diff
@@ -0,0 +1,29 @@
+// RUN: %clang_cc1 -fsyntax-only -verify -pedantic-errors %s
+
+namespace c {
+  extern "C" void main(); // expected-error {{invalid linkage specification 'extern "C"'}}
+}
+extern "C" {
+  int main(); // expected-error {{invalid linkage specification 'extern "C"'}}
+}
+extern "C" int main(); // expected-error {{invalid linkage specification 'extern "C"'}}
+extern "C" struct A { int main(); }; // ok
+
+namespace cpp {
+  extern "C++" int main(); // expected-error {{invalid linkage specification 'extern "C++"'}}
```

Comment:
This one is allowed, https://eel.is/c++draft/basic.start.main#3.4 only disallows `extern "C"` `main`. If you are implementing that, could you add tests for functions (`extern "C" { int main(); }`) and variables (`extern "C" { void* main; }`) named `main`?

Reply:
@MitalAshok Thanks for the feedback. I've added [linkage3.cpp](https://github.com/llvm/llvm-project/pull/101853/files#diff-dc2bba9678021106302e8711c2140ad1e575de50b157537e64b37621e97b9ec9) to cover this case.